plugins/catalog/src/alpha.tsx

```diff
@@ -65,12 +81,203 @@ export const CatalogSearchResultListItemExtension =
+
+/** @alpha */
+export const CatalogEntityKindFilter = createCatalogFilterExtension({
+  id: 'entity.kind',
+  configSchema: createSchemaFromZod(z =>
+    z.object({
+      initialFilter: z.string().default('component'),
+    }),
+  ),
+  loader: async ({ config }) => {
+    const { EntityKindPicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityKindPicker initialFilter={config.initialFilter} />;
+  },
+});
+
+/** @alpha */
+export const CatalogEntityTypeFilter = createCatalogFilterExtension({
+  id: 'entity.type',
+  loader: async () => {
+    const { EntityTypePicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityTypePicker />;
+  },
+});
+
+/** @alpha */
+export const CatalogEntityOwnerFilter = createCatalogFilterExtension({
+  id: 'entity.mode',
+  configSchema: createSchemaFromZod(z =>
+    z.object({
+      mode: z.enum(['owners-only', 'all']).optional(),
+    }),
+  ),
+  loader: async ({ config }) => {
+    const { EntityOwnerPicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityOwnerPicker mode={config.mode} />;
+  },
+});
+
+/** @alpha */
+export const CatalogEntityNamespaceFilter = createCatalogFilterExtension({
+  id: 'entity.namespace',
+  loader: async () => {
+    const { EntityNamespacePicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityNamespacePicker />;
+  },
+});
+
+/** @alpha */
+export const CatalogEntityLifecycleFilter = createCatalogFilterExtension({
+  id: 'entity.lifecycle',
+  loader: async () => {
+    const { EntityLifecyclePicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityLifecyclePicker />;
+  },
+});
+
+/** @alpha */
+export const CatalogEntityProcessingStatusFilter = createCatalogFilterExtension(
+  {
+    id: 'entity.processing.status',
+    loader: async () => {
+      const { EntityProcessingStatusPicker } = await import(
+        '@backstage/plugin-catalog-react'
+      );
+      return <EntityProcessingStatusPicker />;
+    },
+  },
+);
+
+/** @alpha */
+export const CatalogUserListFilter = createCatalogFilterExtension({
+  id: 'user.list',
+  configSchema: createSchemaFromZod(z =>
+    z.object({
+      initialFilter: z.enum(['owned', 'starred', 'all']).default('owned'),
+    }),
+  ),
+  loader: async ({ config }) => {
+    const { UserListPicker } = await import('@backstage/plugin-catalog-react');
+    return <UserListPicker initialFilter={config.initialFilter} />;
+  },
+});
+
+/** @alpha */
+export const CatalogIndexPage = createPageExtension({
+  id: 'catalog',
+  defaultPath: '/catalog',
+  routeRef: rootRouteRef,
+  inputs: {
+    filters: createExtensionInput({
+      element: coreExtensionData.reactElement,
+    }),
+  },
+  loader: async ({ inputs }) => {
+    const { DefaultCatalogPage } = await import('./components/CatalogPage');
+    const filters = inputs.filters.map(filter => filter.element);
```

Comment:
Would that be interesting to be able to change the filter order via configuration?

Reply:
Certainly a tricky one, it's possible right now, but you have to disable and then enable all filters again. Maybe we could try out a design where the catalog plugin can be configured to disable all filter extensions by default? That'd be the most convenient solution for changing the order I think

Reply:
Didn't we reorder extensions in such a way that explicitly mentioned ones come before those that weren't?

Reply:
Oh I see - it applies to previously disabled things only.

https://github.com/backstage/backstage/blob/master/packages/frontend-app-api/src/wiring/parameters.test.ts#L110

In theory this could be expanded to work for enabled things too right?

Reply:
I wonder if that's the case the filter layout component should support users reordering filters through the UI instead and save it in the config or user settings... Thoughts on this other possibility 🙂 ?

Reply:
@freben not really, because then things would start getting reordered when all you want to do is configured them. Would likely need some extra flag like:

```yaml
app:
  extensions:
  - my.yaml.extension:
      reorder: true
```

At a glance I kinda hate it :p

Yep I do think that the answer in some cases might be to make it user configurable, not something we can rely on though imo since it's quite a bit of work